v2/clients/http/deviceprofile.go

```diff
@@ -136,3 +146,13 @@ func (client *DeviceProfileClient) DeviceProfilesByManufacturerAndModel(ctx cont
 	}
 	return res, nil
 }
+
+// DeviceResourceByProfileNameAndResourceName queries the device resource by profileName and resourceName
+func (client *DeviceProfileClient) DeviceResourceByProfileNameAndResourceName(ctx context.Context, profileName string, resourceName string) (res responses.DeviceResourceResponse, edgexError errors.EdgeX) {
+	requestPath := path.Join(v2.ApiDeviceResourceRoute, v2.Profile, url.QueryEscape(profileName), v2.Resource, url.QueryEscape(resourceName))
+	err := utils.GetRequest(ctx, &res, client.baseUrl, requestPath, nil)
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```

Comment:
Missing caching of response. Just need a simple cache added to DeviceProfileClient that  can be a map of map of DeviceResourceResponse indexed by Profile name and then Resource name. I previous WG meeting we agreed no notification required as resources can only be added.

Reply:
> I previous WG meeting we agreed no notification required as resources can only be added.

Got it. Added cache.